### PR TITLE
UI: [VAULT-21435] Message details

### PR DESCRIPTION
**Pull request by kiannaquach**
**Description**
Screenshot 2023-12-28 at 9 38 10 AM

- fixes create and edit date bug
- adds details view

### CODE REVIEW

File: ui/app/serializers/config-ui/message.js [hunk 1]
```diff
@@ -34,9 +48,12 @@ export default class MessageSerializer extends ApplicationSerializer {
       href: json?.link_href || '',
     };
     // using the snapshot startTime and endTime since the json start and end times are null when
-    // it gets to the serialize function.
-    json.start_time = snapshot.record.startTime;
-    json.end_time = snapshot.record.endTime;
+    // it gets to the serialize function. we need to check to see if it's in iso format. if it's not
```
Comment (Monkeychip) [inline ui/app/serializers/config-ui/message.js:51]: These rows of comments are a little confusing. Can you clarify under what circumstances the serializer returns null (If I remember its if the values are coming from default values and have not been set by the user). 

Also, clarify the flow when we would be seeing a string vs data formatted object.

Let me know if that makes sense.

Comment (kiannaquach) [reply ui/app/serializers/config-ui/message.js:51]: Agreed that the comment is confusing! Let me try to refine it :)

File: ui/lib/config-ui/addon/components/messages/page/details.hbs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    {{/if}}
+  </ToolbarActions>
+</Toolbar>
+
+{{#each @message.allFields as |attr|}}
+  {{#if (or (eq attr.name "endTime") (eq attr.name "startTime"))}}
+    <InfoTableRow
+      @label={{capitalize (humanize (dasherize attr.name))}}
+      @value={{if
+        (and (eq attr.name "endTime") (not (get @message attr.name)))
```
Comment (Monkeychip) [inline ui/lib/config-ui/addon/components/messages/page/details.hbs:40]: This is a little hard to read. Maybe add a comment to help someone later down the road understand what is displayed here?

File: ui/lib/config-ui/addon/components/messages/page/details.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * ```
+ * @param {model} message - message model
+ */
+
+export default class MessageDetails extends Component {
+  @service store;
+  @service router;
+  @service flashMessages;
+
+  @task
```
Comment (Monkeychip) [inline ui/lib/config-ui/addon/components/messages/page/details.js:25]: does this need to be a concurrency task? Do we anticipate the backend taking time to delete something and therefore using the `isRunning` value? I suspect the backend can complete this action very quickly and therefor just using a normal ember data action modifier will be sufficient here, no concurrency needed. 

Checkout how we handle `onDelete` in kv or `onDelete` in ldap.

**Review by Monkeychip — APPROVED** [review]
A couple of non-blocking comments. Nice work!

File: ui/lib/config-ui/addon/components/messages/page/details.js (hunk 3 above)
Comment (kiannaquach) [reply ui/lib/config-ui/addon/components/messages/page/details.js:25]: Good point! I don't think the backend will take too long for a delete. Let me update this

**Pull request merged**